docs/reference/builder.md

```diff
@@ -767,6 +767,30 @@ publish the port when running the container, use the `-p` flag on `docker run`
 to publish and map one or more ports, or the `-P` flag to publish all exposed
 ports and map them to high-order ports.
 
+By default, `EXPOSE` assumes TCP. You can also specify UDP:
+
+```none
+EXPOSE 80/udp
+```
+
+To expose on both TCP and UDP, include two lines:
+
+```none
+EXPOSE 80/tcp
+EXPOSE 80/udp
+```
+
+In this case, if you use `-P` with `docker run`, the port will be exposed once
+for TCP and once for UDP. Remember that `-P` uses a random high-ordered host
```

Comment:
perhaps slightly rephrase to prevent using `random` (see the discussion on https://github.com/moby/moby/issues/34310, and PR https://github.com/moby/moby/pull/35102)